### ISSUE TITLE
Kinesis register-stream-consumer UnknownOperationException

### ISSUE THREAD

**Issue by codeasashu** [localstack/localstack#1159]
I am trying to create a consume stream in kinesis. I created my stream using
`awslocal kinesis create-stream --stream-name mylogstream --shard-count 2` 

which worked. Now I am trying to use lambda consumer, following the tutorial [here](https://docs.aws.amazon.com/lambda/latest/dg/with-kinesis.html#services-kinesis-configure), hence I did

`awslocal kinesis register-stream-consumer --consumer-name con1 --stream-arn arn:aws:kinesis:us-east-1:000000000000:stream/mylogstream`  

which resulted in following error:

>An error occurred (UnknownOperationException) when calling the RegisterStreamConsumer operation:

I tried again with `aws` instead of awslocal:

`aws kinesis register-stream-consumer --consumer-name con1 --stream-arn arn:aws:kinesis:us-east-1:000000000000:stream/mylogstream --endpoint-url=http://localhost:4568` 

which again resulted in same error.

I've configured my aws credentials etc. but nothing seems to fix the issue

**Comment by JonFranchi** [comment]
I am getting this exact same error (with zero meaningful details) when trying to call `aws kinesis list-stream-consumers` against localstack.  

It works fine when I point it at a real AWS account, returning even with no consumers:
`{
    "Consumers": []
}`

**Comment by codeasashu** [comment]
I figured the issue. The issue comes down to separate regions for apis being called. Actually, I was using kinesis with region 'us-east-1' while the lambda was being called in region 'us-west-2'(default). 

@JonFranchi Try changing all your regions to 'us-west-2' for now to see if that fixes it

**Comment by Kunamatata** [comment]
I am having the same error. These are the steps I did to replicate the error:
`awslocal kinesis create-stream --stream-name kinesisStream --shard-count 1`
`kinesis describe-stream --stream-name kinesisStream` to get the ARN
`awslocal kinesis register-stream-consumer --consumer-name myLambdaConsumer \
--stream-arn arn:aws:kinesis:us-east-1:000000000000:stream/kinesisStream`
And the error message I get: **An error occurred (UnknownOperationException) when calling the RegisterStreamConsumer operation**

**Comment by JonFranchi** [comment]
Has anybody identified the issue here?  If memory serves, it has to do with the version of kinesis being mocked, but I have pouring over bug tickets and release notes and can't find the answer.  We're again considering using this, but can't if it can't handle enhanced fanout (newer features).

**Comment by khitsik** [comment]
Has anybody solved this issue? I have the same issue

**Comment by whummer** [comment]
Thanks for reporting @codeasashu @JonFranchi @Kunamatata @khitsik . This should be fixed in #1817 - please re-open this issue if the problem persists. Thanks

**Comment by rwinograd** [comment]
Thank you so much for adding this functionality - this seems to now be the only project that has added support for these APIs and supports KCL v2.  I see patching and augmentation at the proxy layer to fix up gaps in kinesalite, moto, etc. Can these types of changes be pushed down to your dependencies?

There is definitely some additional complexity around patching the upstream dependency (kinesalite is a node project versus python, it's a completely different project, different release cycles, etc...) and there are likely many cases where the functionality doesn't belong in the upstream (for example, error injection may not be a feature that is supportable in a consistent way across all dependencies).

However, not everyone can use LocalStack; LocalStack does a great job of presenting a unified interface over a wide range of tools, but the manner in which it pulls packages from binary repositories at runtime means it cannot be used in my deployment environment without significant patching. Pushing the API changes on the upstream project address that pain for a wider audience.

__TLDR:__ If LocalStack is going to continue to rely on other libraries, shouldn't _core_ API implementations in the `proxy_listener` layer be avoided?
